src/build_platform/local_docker.rs

```diff
@@ -70,32 +72,48 @@ impl LocalDocker {
             return;
         }
 
-        // ensure there is enough disk space left before building a new image
-        let docker_path_string = "/var/lib/docker";
-        let root_path_string = "/";
-        let docker_path = Path::new(docker_path_string);
-        let root_path = Path::new(root_path_string);
-
-        // get system info
-        let mut system = sysinfo::System::new_all();
-        system.refresh_all();
-
-        for disk in system.get_disks() {
-            if disk.get_mount_point() == docker_path || disk.get_mount_point() == root_path {
-                let event_details = self.get_event_details();
-                if let Err(e) = check_docker_space_usage_and_clean(
-                    &self.context.docker,
-                    disk,
-                    event_details.clone(),
-                    &*self.logger(),
-                ) {
-                    self.logger.log(EngineEvent::Warning(
-                        event_details,
-                        EventMessage::new(e.to_string(), Some(e.to_string())),
-                    ));
-                }
-                break;
-            };
+        // arbitrary percentage that should make the job anytime
+        const DISK_FREE_SPACE_PERCENTAGE_BEFORE_PURGE: u64 = 20;
+        let mount_points_to_check = vec![Path::new("/var/lib/docker"), Path::new("/")];
+        let mut disk_free_space_percent: u64 = 100;
+
+        let sys_info = sysinfo::System::new_with_specifics(RefreshKind::new().with_disks().with_disks_list());
+        let should_reclaim_space = sys_info.get_disks().iter().any(|disk| {
+            // Check disk own the mount point we are interested in
+            if !mount_points_to_check.contains(&disk.get_mount_point()) {
+                return false;
+            }
+
+            // Check if we have hit our threshold regarding remaining disk space
+            disk_free_space_percent = disk.get_available_space() * 100 / disk.get_total_space();
+            if disk_free_space_percent <= DISK_FREE_SPACE_PERCENTAGE_BEFORE_PURGE {
+                return true;
+            }
+
+            false
+        });
+
+        if !should_reclaim_space {
+            debug!(
+                "Docker skipping image purge, still {} % disk free space",
+                disk_free_space_percent
+            );
+            return;
+        }
+
+        let msg = format!(
+            "Purging docker images to reclaim disk space. Only {} % disk free space, This may take some time",
+            disk_free_space_percent
+        );
+        self.logger
+            .log(EngineEvent::Info(self.get_event_details(), EventMessage::new_from_safe(msg)));
+
+        // Request a purge if a disk is being low on space
+        if let Err(err) = self.context.docker.prune_images() {
+            let msg = format!("Error while purging docker images: {}", err);
+            error!("{}", &msg);
```

Comment:
no need, can be removed

Reply:
I want to put it in error in the engine logs ^^

Reply:
It's done on the next line already, using `logger.log` :)

Reply:
I guess, we can just add a function to log error from a string, it would be a nice add I guess :)